### PR TITLE
Add endpoint to sign_in and sign_out via API

### PR DESCRIPTION
**Pull request by jtapia**
You need to use `/api/v1/spree_user/sign_in` same params `spree_user: { email: 'email', password: 'password' }`

### CODE REVIEW

File: app/controllers/spree/api/v1/user_sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                flash[:success] = Spree.t(:logged_in_succesfully)
+                redirect_back_or_default(after_sign_in_path_for(spree_current_user))
+              }
+              format.js {
+                render json: { user: spree_current_user,
+                               ship_address: spree_current_user.ship_address,
+                               bill_address: spree_current_user.bill_address }.to_json
+              }
+              format.json {
+                render json: { success: true,
```
Comment (softr8) [inline app/controllers/spree/api/v1/user_sessions_controller.rb]: I dont think we need to return success: true, we already have status code, the client should validate that, so with the format.js should be enough

File: app/controllers/spree/api/v1/user_sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            respond_to do |format|
+              format.html {
+                flash.now[:error] = t('devise.failure.invalid')
+                render :new
+              }
+              format.js {
+                render :json => { error: t('devise.failure.invalid') }, status: :unprocessable_entity
+              }
+              format.json {
+                render json: { success: false, error: t('devise.failure.invalid') }
```
Comment (softr8) [inline app/controllers/spree/api/v1/user_sessions_controller.rb]: Same here, we need to use http error codes, this should return either forbidden or unprocessable_entity

File: app/controllers/spree/api/v1/user_sessions_controller.rb (hunk 1 above)
Comment (jtapia) [inline app/controllers/spree/api/v1/user_sessions_controller.rb]: we're using `json` calls, so we need to respond with it, I added the same result as `js` respond

File: app/controllers/spree/api/v1/user_sessions_controller.rb (hunk 2 above)
Comment (jtapia) [inline app/controllers/spree/api/v1/user_sessions_controller.rb]: done

**Comment by softr8** [comment]
what do you think about creating a new action let's say called login_from_mobile and then skip authenticity token only for that action so it is backward compatible with everything else

**Comment by jtapia** [comment]
@softr8 sounds good, doing it right now

**Pull request closed without merging**